frontend/assessments/2/client.js

```diff
@@ -0,0 +1,118 @@
+const socket = io("http://localhost:3001");
+
+socket.on('connect', () => {
+    console.log('Connected to server');
+});
+
+socket.on('disconnect', () => {
+    console.log('Disconnected from server');
+});
+
+const historyDiv = document.querySelector(".history");
+const buyBtn = document.getElementById("buy-btn");
+const sellBtn = document.getElementById("sell-btn");
+const grid = document.querySelector(".grid");
+
+const btnHandler = (buyOrSell) => {
+    let price = parseFloat(document.getElementById("price").textContent);
+    let qty = document.getElementById("qty-input").value;  
+    let currentTime = createdDate = new Date().toLocaleString('en-US', { timeZone: 'Asia/Kolkata' });
+
+    const div = document.createElement('div');
+    div.classList.add("history-div");
+
+    const div1 = document.createElement('div');
+    div1.classList.add('history-price-qty-date');
+
+    const div2 = document.createElement('div');
+    if(buyOrSell == "buy"){
+        div2.classList.add('history-buy-txn');
+        div2.textContent = `Buy`;
+    }
+    else{
+        div2.classList.add('history-sell-txn');
+        div2.textContent = `Sell`;
+    }
+
+    const p1 = document.createElement('p');
+    p1.classList.add("history-price-qty");
+    p1.textContent = `${qty} stocks (Price : ${price}) `;
+
+    const p2 = document.createElement('p');
+    p2.classList.add("history-date");
+    p2.textContent = `${currentTime}`;
+
+
+    div1.appendChild(p1);
+    div1.appendChild(p2);
+    div.appendChild(div1);
+    div.appendChild(div2);
+    historyDiv.appendChild(div);
+
+    socket.emit("save history",div.innerHTML);
```

Comment:
Socket messages, BUY/SELL strings, PORT number etc should be stored in constants